packages/engine/src/gltf/GLTFState.tsx

```diff
@@ -659,7 +658,10 @@ const NodeReactor = (props: { nodeIndex: number; childIndex: number; parentUUID:
       for (const extension in node.extensions.value) {
         const Component = ComponentJSONIDMap.get(extension)
         if (!Component) continue
-        setComponent(entity, Component, node.extensions[extension].get(NO_PROXY_STEALTH))
+
+        const serialized = Component.toJSON(node.extensions[extension].get(NO_PROXY_STEALTH))
```

Comment:
is this going to work for things like colors and anything with a function call?

I think we should lean on explicit schema validation. Maybe that can be implemented in toJSON instead?

Reply:
Shouldn't this extension data only be what is serializable in the json? (No classes or functions)

toJSON is currently a wrapper for SerializeSchema which is explicit schema validation, I'd prefer the schema functions be exposed in as few places as possible, but we could change this to use the validation function directly.